features/lwipstack/LWIPStack.cpp

```diff
@@ -309,14 +309,27 @@ nsapi_error_t LWIP::socket_open(nsapi_socket_t *handle, nsapi_protocol_t proto)
         return NSAPI_ERROR_NO_SOCKET;
     }
 
-    enum netconn_type lwip_proto = proto == NSAPI_TCP ? NETCONN_TCP : NETCONN_UDP;
+    enum netconn_type netconntype;
+    if (proto == NSAPI_TCP) {
+        netconntype = NETCONN_TCP;
+    } else if (proto == NSAPI_UDP) {
+        netconntype = NETCONN_UDP;
+    } else {
+        netconntype = NETCONN_RAW;
+    }
+
 
 #if LWIP_IPV6
     // Enable IPv6 (or dual-stack)
-    lwip_proto = (enum netconn_type)(lwip_proto | NETCONN_TYPE_IPV6);
+    netconntype = (enum netconn_type)(netconntype | NETCONN_TYPE_IPV6);
```

Comment:
Another separate PR addressing LWIPStack

Reply:
features/lwipstack/LWIPStack.cpp change to add NSAPI_RAW support is now submitted as https://github.com/ARMmbed/mbed-os/pull/10978